cmd/gossamer/config.go

```diff
@@ -474,8 +493,13 @@ func setDotGlobalConfigFromFlags(ctx *cli.Context, cfg *dot.GlobalConfig) error
 	cfg.PublishMetrics = ctx.Bool("publish-metrics")
 
 	// check --metrics-address flag and update node configuration
-	if metricsAddress := ctx.GlobalString(MetricsAddressFlag.Name); metricsAddress != "" {
-		cfg.MetricsAddress = metricsAddress
+	if metricsAddress := ctx.String(MetricsAddressFlag.Name); metricsAddress != "" {
+		port, err := strconv.Atoi(metricsAddress)
+		if err != nil {
+			cfg.MetricsAddress = metricsAddress
+		} else {
+			cfg.MetricsAddress = ":" + fmt.Sprint(port)
+		}
```

Comment:
This is supposed to parse as an integer now?

Reply:
The parsing integer is to handle the new alias for MetriceAddressFlag `--prometheus-port` which is used by zombienet.  The original `--metrics-address` flag used a string in the form `:<port>`.  So this new code checks if the value received is an int, than prefix that with a `:`  If the value received is not an int than use the value directly, as the original flag is handled.